src/main/scala/fi/aalto/cs/apluscourses/intellij/actions/REPLAction.scala

```diff
@@ -0,0 +1,64 @@
+package fi.aalto.cs.apluscourses.intellij.actions
+
+import com.intellij.execution.{RunManager, RunManagerEx, RunnerAndConfigurationSettings}
+import com.intellij.openapi.actionSystem.{AnActionEvent, CommonDataKeys}
+import com.intellij.openapi.module.{Module, ModuleUtilCore}
+import com.intellij.openapi.vfs.VirtualFile
+import org.jetbrains.annotations.{NotNull, Nullable}
+import org.jetbrains.plugins.scala.console.actions.RunConsoleAction
+import org.jetbrains.plugins.scala.console.configuration.ScalaConsoleRunConfiguration
+
+import scala.collection.JavaConverters._
+
+class REPLAction extends RunConsoleAction {
+
+  override def actionPerformed(e: AnActionEvent): Unit = {
+    customDoRunAction(e)
+  }
+
+  def checkFileOrFolderIsNull(@Nullable fileOrFolder: VirtualFile): Boolean = {
+    fileOrFolder == null
+  }
+
+  def adjustRunConfigurationSettings(@NotNull module: Module, @NotNull configuration: ScalaConsoleRunConfiguration): Unit = {
+    // adjust the configuration with: name, workDir and module
+    val moduleDirPath = ModuleUtilCore.getModuleDirPath(module)
+    configuration.setWorkingDirectory(moduleDirPath)
+    configuration.setModule(module)
+    configuration.setName("Scala REPL for module: " + module.getName)
+  }
+
+  def customDoRunAction(e: AnActionEvent): Unit = {
```

Comment:
this whole method is a mess, and any proposals on how to improve it are welcome :exclamation:

Reply:
one of the options I see could be cutting it into multiple smaller methods, thus, it will probably become harder to read, as there's gonna be plenty of joggling with same data

Reply:
Not that bad IMO. It's quite straightforward after all. I counted three different paths of execution. I'd suggest either inlining the nested function or  moving it outside of this function.